src/spesh/inline.c

```diff
@@ -1,5 +1,12 @@
 #include "moar.h"
 
+/*
+ * When trying to make a spesh graph to inline from an unspecialized static
+ * frame, we run MVM_spesh_optimize, which can lead to another attempt to
+ * create a graph the same way. Don't do that!
+ */
+#define SPESH_UNSPECIALIZED_INLINE_RECURSION_DEPTH_LIMIT 3
```

Comment:
Feels like there are still a lot of sensible numbers between 3 and infinity. In other words, this limit seems quite low. How did you come up with it?